api/src/utils/transformations.ts

```diff
@@ -1,33 +1,59 @@
-import type { File, Transformation, TransformationParams } from '../types/index.js';
+import type { File, Transformation, TransformationFormat, TransformationSet } from '../types/index.js';
 
-export function resolvePreset(input: TransformationParams, file: File): Transformation[] {
-	const transforms = input.transforms ? [...input.transforms] : [];
+export function resolvePreset({ transformationParams, acceptFormat }: TransformationSet, file: File): Transformation[] {
+	const transforms = transformationParams.transforms ? [...transformationParams.transforms] : [];
 
-	if (input.format || input.quality) {
+	if (transformationParams.format || transformationParams.quality) {
 		transforms.push([
 			'toFormat',
-			input.format || (file.type!.split('/')[1] as any),
+			getFormat(file, transformationParams.format, acceptFormat),
 			{
-				quality: input.quality ? Number(input.quality) : undefined,
+				quality: transformationParams.quality ? Number(transformationParams.quality) : undefined,
 			},
 		]);
 	}
 
-	if (input.width || input.height) {
+	if (transformationParams.width || transformationParams.height) {
 		transforms.push([
 			'resize',
 			{
-				width: input.width ? Number(input.width) : undefined,
-				height: input.height ? Number(input.height) : undefined,
-				fit: input.fit,
-				withoutEnlargement: input.withoutEnlargement ? Boolean(input.withoutEnlargement) : undefined,
+				width: transformationParams.width ? Number(transformationParams.width) : undefined,
+				height: transformationParams.height ? Number(transformationParams.height) : undefined,
+				fit: transformationParams.fit,
+				withoutEnlargement: transformationParams.withoutEnlargement
+					? Boolean(transformationParams.withoutEnlargement)
+					: undefined,
 			},
 		]);
 	}
 
 	return transforms;
 }
 
+function getFormat(
+	file: File,
+	format: TransformationSet['transformationParams']['format'],
+	acceptFormat: TransformationSet['acceptFormat']
+): TransformationFormat {
+	const fileType = file.type!.split('/')[1] as TransformationFormat;
+
+	if (format) {
+		if (format !== 'auto') {
+			return format;
+		}
+
+		if (acceptFormat) {
+			return acceptFormat;
+		}
+
+		if (fileType !== 'png') {
+			return 'jpg';
+		}
```

Comment:
> I wonder if there are other formats which should not get converted to `jpg`. We could also come up with a map like:
> 
> ```
> png  -> png
> avif -> png
> tiff -> png?
> 
> else -> jpg
> ```

first of all. ty for working on this! You are awesome!

SVG, GIF and WebP support transparency as well. But I mean... who uses gif for transparency anymore lol

Reply:
Pascal and I were just discussing this on Discord :) 


> So instead of hardcoded check for png -> jpg, should we have a map of file types that might have transparency, and check against that?
> I'm pretty sure webp and avif also support transparency, so I'm starting to think we might want to default to png instead of jpg for files that aren't a jpg already
> So try avif -> try webp -> default to current type

Reply:
> Pascal and I were just discussing this on Discord :)
> 
> > So instead of hardcoded check for png -> jpg, should we have a map of file types that might have transparency, and check against that?
> > I'm pretty sure webp and avif also support transparency, so I'm starting to think we might want to default to png instead of jpg for files that aren't a jpg already
> > So try avif -> try webp -> default to current type

We could even instead of converting all transparency to PNG we could convert them to WebP. It retains the transparency and is smaller than a png usually. 

But a map would be cool, it allows anyone to switch between if needed for their use case and allows for future edits of new extensions to be added. 

Great ideas you two!

Reply:
> We could even instead of converting all transparency to PNG we could convert them to WebP. It retains the transparency and is smaller than a png usually.

Yeah! The default to png would only apply when `format=auto` is used, and the browser's accept header doesn't include avif or webp 🙂 Right now, it defaults to `jpg`, which means that `png`s are transformed to `jpg` when the accept header doesn't include avif/webp, which in turn causes the black backgrounds, as jpg doesn't have transparency 👍🏻

Reply:
Result with images before & after this PR:

<img width="800" src="https://github.com/directus/directus/assets/5363448/99c27285-ce73-4ddf-bc28-2a636c101bf3">

Note: GIFs are never converted.